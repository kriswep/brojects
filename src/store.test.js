/**
 * Test store
 */

import configureStore from './store';

describe('configureStore', () => {
  describe('the store', () => {
    it('should configure a redux store', () => {
      const store = configureStore(undefined);
      expect(typeof store).toBe('object');
    });

    it('should configure store with proper inital state', () => {
      const store = configureStore(undefined);
      expect(store.getState()).toMatchSnapshot();
    });

    it('should configure store with given initial state', () => {
      const store = configureStore({
        auth: {
          authToken: 'initialToken',
        }
      });
      expect(store.getState()).toMatchSnapshot();
    });
  });
});