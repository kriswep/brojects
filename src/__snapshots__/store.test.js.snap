exports[`configureStore the store should configure store with given initial state 1`] = `
Object {
  "auth": Object {
    "authToken": "initialToken",
  },
  "cards": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "columnData": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "columns": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "projects": Object {
    "currentRepo": false,
    "data": Array [],
    "error": false,
    "loading": false,
  },
  "repos": Object {
    "currentRepo": false,
    "data": Array [],
    "error": false,
    "loading": false,
  },
}
`;

exports[`configureStore the store should configure store with proper inital state 1`] = `
Object {
  "auth": Object {
    "authtoken": false,
  },
  "cards": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "columnData": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "columns": Object {
    "data": Object {},
    "error": false,
    "loading": false,
  },
  "projects": Object {
    "currentRepo": false,
    "data": Array [],
    "error": false,
    "loading": false,
  },
  "repos": Object {
    "currentRepo": false,
    "data": Array [],
    "error": false,
    "loading": false,
  },
}
`;
